rocket_twin/systems/rocket/rocket.py

```diff
@@ -24,54 +16,44 @@ class Rocket(System):
     ------
     """
 
-    def setup(self):
-        self.add_child(ControllerCoSApp("controller"))
-        self.add_child(Tank("tank"), pulling=["w_in", "weight_max", "weight_prop"])
-        self.add_child(Engine("engine"))
-        self.add_child(NoseGeom("nose"))
-        self.add_child(TubeGeom("tube"))
-        self.add_child(WingsGeom("wings"))
-        self.add_child(
-            OCCGeometry(
-                "geom",
-                shapes=["tank_s", "engine_s", "nose_s", "tube_s", "wings_s"],
-                properties=["tank", "engine", "nose", "tube", "wings"],
+    def setup(self, n_stages=1):
+
+        shapes, properties, forces = ([None] * n_stages for i in range(3))
+
+        for i in range(1, n_stages + 1):
+            nose = False
+            wings = False
+            if i == 1:
+                wings = True
+            if i == n_stages:
+                nose = True
+
+            self.add_child(
+                Stage(f"stage_{i}", nose=nose, wings=wings),
+                pulling={
+                    "w_in": f"w_in_{i}",
+                    "weight_max": f"weight_max_{i}",
+                    "weight_prop": f"weight_prop_{i}",
+                },
             )
-        )
-        self.add_child(
-            Dynamics(
-                "dyn",
-                forces=["thrust"],
-                weights=["weight_rocket"],
-            ),
-            pulling=["a"],
-        )
+            shapes[i - 1] = f"stage_{i}_s"
+            properties[i - 1] = f"stage_{i}"
+            forces[i - 1] = f"thrust_{i}"
```

Comment:
could you explain in the doctrings these values ?

Reply:
Ok!